classes/security/Validation.php

```diff
@@ -474,6 +482,65 @@ public static function canAdminister($administeredUserId, $administratorUserId)
         // There were no conflicting roles. Permit administration.
         return true;
     }
+
+    /**
+     * Get the user's administration level
+     *
+     * @param int   $administeredUserId     User ID of user to potentially administer
+     * @param int   $administratorUserId    User ID of user who wants to do the administrating
+     * 
+     * @return int The authorized adminstration level 
+     */
+    public static function getAdministrationLevel($administeredUserId, $administratorUserId)
+    {
+        $roleDao = DAORegistry::getDAO('RoleDAO'); /** @var RoleDAO $roleDao */
```

Comment:
`RoleDAO` has been largely replaced by the UserGroup repository class and collector; I'd suggest rewriting this to use those instead. See notes below.

Reply:
Removed and fully moved to `UserGroup Repository` .